docs/dg/learningBasics.md

```diff
@@ -0,0 +1,416 @@
+
+**Command Line Argument Parser**
+
+1. [Generate a report locally](../ug/generatingReports.html#generating-reports-locally).
+2. Open a generated JSON file. For example, you can open `summary.json` of the generated report.
+3. You should see that the content in the JSON is compactly organized in 1 single line. This may affect readability for developer who wants to investigate the content in the JSON file.
+
+**Your Task**
+
+Add a new CLI argument `--use-json-pretty-printing`, such that when a user runs a command `java -jar RepoSense.jar --repos LIST_OF_REPO_URLS --view --use-json-pretty-printing`, the JSON file will be printed in a more readable way.
+
+<panel header="Hint 1">
+
+Try to understand the parsing process of the Command Line Arguments, which starts from `ArgsParser.parse(args)` in `RepoSense.java` and uses `ArgumentParser` to convert an argument string into a Java Object.
+
+Therefore, the first step you can take is to add the following to `ArgsParser`.
+
+```
+public static final String[] JSON_PRINT_MODE_FLAGS = new String[]{"--use-json-pretty-printing", "-j"};
+```
+
+In `getArgumentParser` method, add the following content.
+
+```
+parser.addArgument(JSON_PRINT_MODE_FLAGS)
+    .dest(JSON_PRINT_MODE_FLAGS[0])
+    .action(Arguments.storeTrue())
+    .help("A flag to use json pretty printing when generating the json files.");
+```
+</panel>
+
+<panel header="Hint 2">
+
+After the step in hint 1, the argument is captured by `ArgumentParser`. Now change the  `CliArguments`, `ConfigCliArguments`, `parse` method in `ArgsParser` to make the return result of `parse` include the new argument. 
+
+Add the following content to `CliArguments`.
+
+```
+protected boolean isPrettyPrintingUsed;
+
+public boolean isPrettyPrintingUsed() {
+        return isPrettyPrintingUsed;
+}
+```
+
+In the [constructor](https://github.com/reposense/RepoSense/blob/9125d4b6bea1e5bb6329cb8f7c476ea18fed3cea/src/main/java/reposense/model/ConfigCliArguments.java#L26-L52) of `ConfigCliArguments`, add `isPrettyPrintingUsed` as a new argument of the method, add the following content to the method body.
+
+```
+this.isPrettyPrintingUsed = isPrettyPrintingUsed;
+```
+
+In the `parse` method, replace the [section](https://github.com/reposense/RepoSense/blob/9125d4b6bea1e5bb6329cb8f7c476ea18fed3cea/src/main/java/reposense/parser/ArgsParser.java#L303-L306) with the following content to get `isJsonPrettyPrintingUsed` from `ArgmentParser` and pass it to `ConfigCliArguments`.
+
+```
+boolean isJsonPrettyPrintingUsed = results.get(JSON_PRINT_MODE_FLAGS[0]);
+
+return new ConfigCliArguments(configFolderPath, outputFolderPath, assetsFolderPath, sinceDate, untilDate,
+        isSinceDateProvided, isUntilDateProvided, numCloningThreads, numAnalysisThreads, formats,
+        shouldIncludeLastModifiedDate, shouldPerformShallowCloning, isAutomaticallyLaunching,
+        isStandaloneConfigIgnored, zoneId, isJsonPrettyPrintingUsed);
+``` 
```

Comment:
Try to minimize giving the exact code, to reduce the risk of this document going out of sync with the actual code.

Reply:
Thanks for the suggestion. I will try to eliminate the code replacement and restrict the presence of such code section only when it is a pure addition